cmd/mistryd/job.go

```diff
@@ -207,3 +208,74 @@ func (j *Job) String() string {
 		"{project=%s params=%s group=%s id=%s}",
 		j.Project, j.Params, j.Group, j.ID[:7])
 }
+
+// CloneSrcPath Determine if we should use the build cache and return the path that
+// should be used.
+//
+// Using the build cache should happen when
+// 1. the job is invoked with a group
+// 2. the symlink pointing to the latest build is valid
+func (j *Job) CloneSrcPath(log *log.Logger) string {
```

Comment:
As a future idea, we should probably add a `logger` field to `Job{}` and use it throughout `Work()` and this function too.